azure-iot-device/azure/iot/device/common/mqtt_transport.py

```diff
@@ -652,15 +674,38 @@ def complete_operation(self, mid):
                 # fully expected.  QOS=1 means we might get 2 PUBACKs
                 logger.debug("No callback set for MID: {}".format(mid))
 
+    def fail_all_subs_and_unsubs(self, error_cls, error_msg):
```

Comment:
This solution is a little over-fitted to solving the problem in question, I admit, but when I make the transport async I'm going to have to re-implement a lot of this anyway. For now, this implementation produces the desired behavior, and that's good enough IMO.